wicket-core/src/main/java/org/apache/wicket/Page.java

```diff
@@ -1012,4 +1013,21 @@ public final boolean wasRendered(Component component)
 	{
 		return renderedComponents != null && renderedComponents.contains(component);
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		Page that = (Page) o;
+		return autoIndex == that.autoIndex && numericId == that.numericId;
```

Comment:
you are right, that `autoIndex` is not good for the method as it can change.
But so does `renderCount`.
So I change the method to use `numericId` only.